terraform/stacks/main/outputs.tf

```
@@ -751,19 +751,19 @@ output "csb" {
   sensitive   = true
   value = {
     ecr_user = {
-      username               = module.csb_broker[0].ecr_user_username
-      access_key_id_curr     = module.csb_broker[0].ecr_user_access_key_id_curr
-      secret_access_key_curr = module.csb_broker[0].ecr_user_secret_access_key_curr
-      access_key_id_prev     = module.csb_broker[0].ecr_user_access_key_id_prev
-      secret_access_key_prev = module.csb_broker[0].ecr_user_secret_access_key_prev
+      username               = length(module.csb_broker[0]) > 0 ? module.csb_broker[0].ecr_user_username : ""
```

```suggestion
      username               = length(module.csb_broker) > 0 ? one(module.csb_broker).ecr_user_username : ""
```

`module.csb_broker[0]` would be an object itself, not an array, I think? The `one` thing I think is just nice, but less important

---

Good catch, thanks. It looks like I can simplify this even further to just `one(module.csb_broker[*].property)...` since `one` returns null if the collection contains no elements.